compiler/src/iree/compiler/Utils/test/TileSizeSelectionTest.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "iree/compiler/Codegen/Common/TileSizeSelection.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenDialect.h"
+
+using namespace mlir::iree_compiler;
+using namespace llvm;
+using namespace mlir;
+using namespace testing;
+
+TEST(TileSizeSelection, Basic) {
```

Comment:
nit: Basic does not provide any information to me. How about rename it to `MaxNumTileLevels`, since the test is targeting below snippet?

https://github.com/openxla/iree/blob/f755b422892bfb8ebb2ee8258f9aac140cddbeb4/compiler/src/iree/compiler/Codegen/Common/TileSizeSelection.cpp#L45-L49

Reply:
> nit: Basic does not provide any information to me.

I agree that the current name is not particularly useful. This is really only intended as a smoke test, so perhaps `Basic` -> `SmokeTest`?

Note that `MaxNumTileLevels` is `private` -  it cannot be accessed from these tests. Instead, I am defining `const unsigned kMaxNumTilingLevels = 7;` further down. 

I could add an accessor to `TilingConfig`:
```cpp
static unsigned getMaxNumTilingLevels() {
  return TilingLevel::MaxNumTileLevels;
}
```` 
and then use it 
```cpp
unsigned kMaxNumTilingLevels = getMaxNumTileLevels()
``` 

However, that would be added to facilitate testing.

WDYT? You've been working on this much longer than me and I'm happy to follow your suggestions.